src/lib/list.c

```diff
@@ -324,18 +324,23 @@ int list_strcmp(const void *a, const void *b)
 struct list *list_deduplicate(struct list *list, comparison_fn_t comparison_fn, list_free_data_fn_t list_free_data_fn)
 {
 	struct list *iter = NULL;
-	void *item1, *item2;
+	void *item1, *item2 = NULL;
 
-	list = list_head(list);
+	iter = list = list_head(list);
 
-	for (iter = list; iter && iter->next; iter = iter->next) {
+	while (iter && iter->next) {
```

Comment:
+1

Can you send bug fixes that aren't related to this PR in separate PRs, so we can merge them faster?

Reply:
ok, I will submit a new PR cherry-picking those